File: logica-general/ejercicio-012/resoluciones/lucas-pajarito/pinpong.js
const jugadores = [
    {
        nombre: "Carlos",
        puntos: 21,
        setsGanados: 2
    },
    {
        nombre: "Andrés",
        puntos: 18,
        setsGanados: 1
    },
    {
        nombre: "María",
        puntos: 25,
        setsGanados: 3
    },
    {
        nombre: "Laura",
        puntos: 19,
        setsGanados: 2
    },
    {
        nombre: "José",
        puntos: 16,
        setsGanados: 1
    }
];

function compararJugadores(jugadores) {

    let mejorJugador = jugadores[0];

    jugadores.forEach(jugador => {

        if (jugador.puntos > mejorJugador.puntos) {
            mejorJugador = jugador;
        }

    });

    console.log("=== MEJOR JUGADOR DEL TORNEO ===");
    console.log(`Nombre: ${mejorJugador.nombre}`);
    console.log(`Puntos: ${mejorJugador.puntos}`);
    console.log(`Sets ganados: ${mejorJugador.setsGanados}`);

}

compararJugadores(jugadores);

File: logica-general/ejercicio-032/resoluciones/lucas-pajarito/pinpong.py
jugadores = [
    {
        "nombre": "Juan",
        "puntos": 21
    },
    {
        "nombre": "Pedro",
        "puntos": 18
    },
    {
        "nombre": "Luis",
        "puntos": 25
    },
    {
        "nombre": "Carlos",
        "puntos": 16
    },
    {
        "nombre": "Miguel",
        "puntos": 22
    }
]


def comparar_jugadores(jugadores):

    print("=== COMPARACIÓN DE JUGADORES ===")

    mayor = jugadores[0]

    for jugador in jugadores:

        print(f'Jugador: {jugador["nombre"]}')
        print(f'Puntos: {jugador["puntos"]}')
        print("----------------------------")

        if jugador["puntos"] > mayor["puntos"]:
            mayor = jugador

    print(f'El jugador con mayor puntuación es {mayor["nombre"]} con {mayor["puntos"]} puntos.')


comparar_jugadores(jugadores)